### PR TITLE
Add return empty Instances in store.LoadInstances if not found

### LINKED ISSUE #642: store.LoadInstances should take into account the zero-instance case
Currently `LoadInstances` returns an error in this case, when it could instead return a valid `Instances` struct with zero instances.

Here's the current implementation:
https://github.com/mattermost/mattermost-plugin-jira/blob/830cc03eaf7338e02bb9c40e26d8997ad414de6e/server/kv.go#L534-L543

_Originally posted by @mickmister in https://github.com/mattermost/mattermost-plugin-jira/pull/641#discussion_r474795820_

### PR DESCRIPTION
**Pull request by vladimirdotk**
#### Summary
- Add return empty Instances in store.LoadInstances if not found

#### Ticket Link
Fixes #642

### CODE REVIEW

**Comment by jasonblais** [comment]
@mickmister and @jfrerich quick reminder to help with reviews :)

**Review by mickmister — APPROVED** [review]
Thanks @vladimirdotk! Great work :+1:

**Review by jfrerich — APPROVED** [review]
Thanks for the PR, @vladimirdotk!

**Comment by jasonblais** [comment]
@DHaussermann kind reminder to help with reviews :)

File: server/kv.go [hunk 1]
```diff
@@ -536,6 +537,9 @@ func (store *store) DeleteInstance(id types.ID) error {
 func (store *store) LoadInstances() (*Instances, error) {
 	kv := kvstore.NewStore(kvstore.NewPluginStore(store.plugin.API))
 	vs, err := kv.ValueIndex(keyInstances, &instancesArray{}).Load()
+	if errors.Cause(err) == kvstore.ErrNotFound {
+		return &Instances{}, nil
```
Comment (mickmister) [inline server/kv.go:541]: There is a panic happening due to this `Instances` not being properly initialized. Can we change this line to the following?

```suggestion
		return NewInstances(), nil
```

**Review by mickmister — CHANGES_REQUESTED** [review]
Thanks @vladimirdotk! I have one request on initializing the `Instances` struct

**Review by mickmister — APPROVED** [review]
Great work @vladimirdotk! Thanks for your contribution!!

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Install issue is now resolved! 
- When the list command is run and there is nothing to list, I now see a much friendlier `(none installed)` as opposed to the previous `Failed to load known Jira instances: instances/v3: not found`
- Confirmed in system logs that the error no longer occurs when there are no instances and the command is run to retrieve them
- Regression tested installing and connecting to an instance
LGTM!

Thanks @vladimirdotk for this fix. Much appreciated!

**Pull request merged**